app/views/Login/index.tsx

```diff
@@ -15,33 +26,108 @@ import {
     requiredStringCondition,
     useForm,
 } from '@togglecorp/toggle-form';
-import {
-    CombinedError,
-    gql,
-} from 'urql';
 
-import UserContext from '#base/context/UserContext';
 import Button from '#components/Button';
+import ButtonLayout from '#components/ButtonLayout';
 import TextInput from '#components/TextInput';
-import { useLoginMutation } from '#generated/types/graphql';
 import useAlert from '#hooks/useAlert';
 import mapSwipeLogo from '#resources/images/mapswipe-logo.svg';
-import {
-    alertCombinedError,
-    checkAndAlertGraphQLResultError,
-} from '#utils/error';
 
 import styles from './styles.module.css';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const LOGIN_MUTATION = gql`
-mutation Login($username: String!, $password: String!) {
-    login(username: $username, password: $password) {
-        id
-        displayName
+const APP_ENV = import.meta.env.APP_ENVIRONMENT;
+const COOKIE_NAME = `MAPSWIPE-${APP_ENV}-CSRFTOKEN`;
+const REST_ENDPOINT = import.meta.env.APP_REST_API_DOMAIN;
+const cookies = new Cookies();
+
+async function loginUsingFirebaseToken({ token }: { token: string }) {
+    try {
+        const response = await fetch(`${REST_ENDPOINT}/firebase-auth/`, {
+            method: 'POST',
+            credentials: 'include',
+            headers: {
+                'Content-Type': 'application/json',
+                'X-CSRFToken': cookies.get(COOKIE_NAME),
+            },
+            body: JSON.stringify({ token }),
+        });
+
+        if (!response.ok || response.status !== 200) {
```

Comment:
what if we get 2xx codes? we might not need to check for 200 as we are reading ok